collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -196,18 +206,37 @@ public void onClick(View v) {
     @Override
     protected void onResume() {
         super.onResume();
-        viewModel.resume();
+        mainMenuViewModel.resume();
 
         setButtonsVisibility();
         invalidateOptionsMenu();
     }
 
     private void setButtonsVisibility() {
-        reviewDataButton.setVisibility(viewModel.shouldEditSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        sendDataButton.setVisibility(viewModel.shouldSendFinalizedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        viewSentFormsButton.setVisibility(viewModel.shouldViewSentFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        getFormsButton.setVisibility(viewModel.shouldGetBlankFormButtonBeVisible() ? View.VISIBLE : View.GONE);
-        manageFilesButton.setVisibility(viewModel.shouldDeleteSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        reviewDataButton.setVisibility(mainMenuViewModel.shouldEditSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        sendDataButton.setVisibility(mainMenuViewModel.shouldSendFinalizedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        viewSentFormsButton.setVisibility(mainMenuViewModel.shouldViewSentFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        getFormsButton.setVisibility(mainMenuViewModel.shouldGetBlankFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+        manageFilesButton.setVisibility(mainMenuViewModel.shouldDeleteSavedFormButtonBeVisible() ? View.VISIBLE : View.GONE);
+    }
+
+    @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        final MenuItem projectsMenuItem = menu.findItem(R.id.projects);
+
+        ProjectIconView projectIconView = (ProjectIconView) projectsMenuItem.getActionView();
+        projectIconView.setProject(currentProjectViewModel.getCurrentProject().getValue().getValue());
+        projectIconView.setOnClickListener(v -> onOptionsItemSelected(projectsMenuItem));
+
+        currentProjectViewModel.getCurrentProject().getValue().consume();
+        currentProjectViewModel.getCurrentProject().observe(this, project -> {
+            if (!project.isConsumed()) {
+                invalidateOptionsMenu();
+                ToastUtils.showLongToast(getString(R.string.switched_project, project.getValue().getName()));
```

Comment:
I think we can avoid using `Consumable` here by just moving this toast call in `ProjectSettingsDialog` right? It seems like the `Consumable` ends up being awkward as you have to call `consume` to deal with the "initial" value.

Reply:
but still `invalidateOptionsMenu();` would be called for no reason. Not harmful but...

Reply:
Personally that feels fine to me. We've done that in other places (see `BlankFormListMenuDelegate` for example). If we're worried about a double render we could skip calling it the first time, but it doesn't feel necessary to me.

Reply:
Fixed.